src/core/ext/filters/client_channel/client_channel.cc

```diff
@@ -1990,6 +2042,16 @@ static void recv_trailing_metadata_ready(void* arg, grpc_error* error) {
   }
   // Not retrying, so commit the call.
   retry_commit(elem, retry_state);
+  // Now that the try is committed, give the trailer to the lb policy as needed
+  if (calld->pick.recv_trailing_metadata_ready != nullptr) {
+    GPR_ASSERT(calld->pick.recv_trailing_metadata != nullptr);
```

Comment:
See above -- we probably want to allow LB policies to set the callback without requesting the metadata, so we should allow this to be null.